lib/cchain/provider/abci.go

```diff
@@ -373,11 +373,11 @@ func searchOffsetInHistory(ctx context.Context, client rpcclient.Client, cl atyp
 		endHeightIndex = queryHeight
 		lookback *= 2
 		if queryHeight < lookback {
-			startHeightIndex = 1
-			break
+			// Query from the start, but don't break out yet -- we need to find the earliest height that we have state for
+			queryHeight = 1
+		} else {
+			queryHeight -= lookback
 		}
```

Comment:
Good catch, thank you. I don't think we want to use 0, since that defaults to using the `latest` height. So I think what we want is actually:

```
		if queryHeight <= lookback {
			// Query from the start, but don't break out yet -- we need to find the earliest height that we have state for
			queryHeight = 1
		} else {
			queryHeight -= lookback
		}
```

and we can do this at the top of the loop.